ui/app/src/components/GitManagement/GitManagement.tsx

```diff
@@ -168,6 +194,21 @@ export function GitManagement(props: GitManagementProps) {
     fetchRepoStatus();
   }, [fetchRepoStatus]);
 
+  useEffect(() => {
+    const handleBeforeUpload = (event) => {
+      event.preventDefault();
+      return (event.returnValue = '');
+    };
+    if (repoStatusConflictDialog.open && repoStatus && !repoStatus.clean) {
+      window.onbeforeunload = handleBeforeUpload;
+    } else {
+      window.onbeforeunload = null;
+    }
+    return () => {
+      window.onbeforeunload = null;
+    };
+  }, [repoStatus, repoStatusConflictDialog.open]);
```

Comment:
Wouldn't this show the dialog for everyone landing on Git? the idea was use the promise on pull success to show only for the puller that first received the conflicts